Mathlib/Algebra/Squarefree/UniqueFactorizationDomain.lean

```diff
@@ -94,3 +94,17 @@ theorem squarefree_mul_iff :
       rw [mul_mul_mul_comm] at contra; exact dvd_of_mul_right_dvd contra
     exact ⟨hy.dvd_of_squarefree_of_mul_dvd_mul_left contra,
       hx.dvd_of_squarefree_of_mul_dvd_mul_right contra⟩
+
+lemma exists_squarefree_dvd_pow_of_ne_zero (hx : x ≠ 0) :
+    ∃ (y : R) (n : ℕ), Squarefree y ∧ x ∣ y ^ n := by
```

Comment:
This can be [strengthened](https://github.com/leanprover-community/mathlib4/compare/DecompositionMonoid?expand=1#diff-8088736f0d4b8a7f07a031cbe91d9579d9dc9c4057fa9008eb51a25c5c57ec43R106-R119) to require that y divides x.

I started the DecompositionMonoid refactoring in that branch, and I think every lemma in this file can be generalized to DceompositionMonoid (and two only require CancelCommMonoidWithZero as you can see). However this new lemma can't be generalized, since (assuming DecompositionMonoid) it implies factorizability into squarefree elements, which is the subject of Proposition 3.4 in [the paper](https://link.springer.com/article/10.1007/s00233-019-10022-3). Note that DecompositionMonoid + factorizability into irreducibles would imply UFM right away.

Reply:
Having Bezout in the name is nice; IsCoprime is equivalent to IsRelPrime in Bezout rings (`IsBezout`), and most results in Mathlib.RingTheory.PrincipalIdealDomain can be generalized to Bezout rings (but maybe the correct approach is to replace IsCoprime by IsRelPrime and use the equivalence when necessary ...). There is `IsBezout.gcd` but maybe it's better to allow a separate `GCDMonoid` instance and use that `gcd` so the lemmas won't just apply to the particular choice of `gcd`/generators.

Indeed most lemmas about IsRelPrime has `coprime` in their names instead. I think we have to use a new name (and IsRelPrime) is my choice, since IsCoprime is widely used and we'd like to deprecate it rather than renaming it right away. I won't rename the lemmas so that they have the right names when we later rename IsRelPrime back to IsCoprime. (But do we need to deprecate IsRelPrime then? Maybe it's better to rename IsCoprime right now ...)

I still think IsComaximal is a reasonable name, since `IsCoprime x y` is equivalent to the [comaximality](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/coprime.20ideal/near/395125656) (currently with the unintuitive name [Codisjoint](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Disjoint.html#Codisjoint)) of the ideals they generate, and also to the comaximality of the basic opens in the spectrum.